### PR TITLE
feat: add sortNodes function in brush config

### PR DESCRIPTION
**Pull request by jjxie**
Add a  function called `sortNodes` in brush config.

```
{
    consume: [
      {
        data,
        context: 'selection',
        mode: 'and',
        style: {
          inactive: {},
          active: {},
        },
      },
    ],
    sortNodes: (obj) => {
      return sortedNodes;
    },
  };
```

### CODE REVIEW

**Comment by jjxie** [comment]
> Im guessing there is some very specific use case in mind with this change? However do consider making the API a bit more flexible, ex: by allowing the user to configure a sort callback function instead.

Yea, it is mainly to fix scatterplot point highlighting issue and don't want to effect other charts. I will check how to make it more flexible.
Screenshot 2022-02-07 at 16 33 23

**Comment by cbt1** [comment]
@jjxie I see. So my point is that down the road we might need to fix some other sorting related issue for other specific case. In which case we would need to add another property/option we makes the code and the API more complex. 

So try avoid that by providing some other means of solving, ex. by using a sort callback.

File: packages/picasso.js/src/core/component/brushing.js [hunk 1]
```diff
@@ -141,7 +141,11 @@ export function styler(obj, { context, data, style, filter, mode }) {
   const onUpdate = () => {
     const changed = update();
     if (changed) {
-      obj.renderer.render(obj.nodes);
+      if (typeof obj.config.sortNodes === 'function') {
+        obj.config.sortNodes(obj);
```
Comment (quanho) [inline packages/picasso.js/src/core/component/brushing.js:145]: It should be obj.renderer.render(obj.config.sortNodes(obj));

Comment (jjxie) [reply packages/picasso.js/src/core/component/brushing.js]: Changed `obj.renderer.render` back to be in Picasso

File: packages/picasso.js/src/core/component/brushing.js [hunk 2]
```diff
@@ -141,7 +141,11 @@ export function styler(obj, { context, data, style, filter, mode }) {
   const onUpdate = () => {
     const changed = update();
     if (changed) {
-      obj.renderer.render(obj.nodes);
+      if (typeof obj.config.sortNodes === 'function') {
+        obj.renderer.render(obj.config.sortNodes(obj));
```
Comment (veinfors) [inline packages/picasso.js/src/core/component/brushing.js:145]: Looks good to me. But I think we should add this `sortNodes` function in the Api spec.

**Review by veinfors — APPROVED** [review]

**Review by cbt1 — APPROVED** [review]

**Pull request merged**